lib/operation.js

```diff
@@ -787,24 +787,28 @@ function linear (a, b) {
  *     // With this example input, a sepia filter has been applied
  *   });
  *
- * @param {Array<Array<number>>} inputMatrix - 3x3 Recombination matrix
+ * @param {Array<Array<number>>} inputMatrix - 3x3 Recombination matrix or 4x4 Recombination matrix
  * @returns {Sharp}
  * @throws {Error} Invalid parameters
  */
 function recomb (inputMatrix) {
-  if (!Array.isArray(inputMatrix) || inputMatrix.length !== 3 ||
-      inputMatrix[0].length !== 3 ||
-      inputMatrix[1].length !== 3 ||
-      inputMatrix[2].length !== 3
-  ) {
-    // must pass in a kernel
+  if (!Array.isArray(inputMatrix)) {
+    throw new Error('Invalid recombination matrix');
+  }
+
+  const length = inputMatrix.length;
+  if (length !== 3 && length !== 4) {
     throw new Error('Invalid recombination matrix');
   }
-  this.options.recombMatrix = [
-    inputMatrix[0][0], inputMatrix[0][1], inputMatrix[0][2],
-    inputMatrix[1][0], inputMatrix[1][1], inputMatrix[1][2],
-    inputMatrix[2][0], inputMatrix[2][1], inputMatrix[2][2]
-  ].map(Number);
+
+  for (const row of inputMatrix) {
+    if (row.length !== length) {
+      throw new Error('Invalid recombination matrix');
+    }
+  }
+
+  this.options.recombMatrix = inputMatrix.flat().map(Number);
```

Comment:
Excellent use of `flat()` here, I suspect it wasn't fully-available when this feature was first added. Perhaps this also provides a simpler way to validate the matrix length, i.e. must be either 9 or 16?

Reply:
fixed